File: static/guest_registration.js
document.addEventListener("DOMContentLoaded", function () {
  loadGuestRegistrationMeta();

  const form = document.getElementById("guestRegistrationForm");
  const clearBtn = document.getElementById("guestClearBtn");

  form.addEventListener("submit", function (e) {
    e.preventDefault();

    const payload = {
      firstName: document.getElementById("firstName").value.trim(),
      lastName: document.getElementById("lastName").value.trim(),
      email: document.getElementById("email").value.trim(),
      phoneNumber: document.getElementById("phoneNumber").value.trim(),
      dateOfBirth: document.getElementById("dateOfBirth").value,
      passportId: document.getElementById("passportId").value.trim(),
      highFloor: document.getElementById("highFloor").checked,
      quietRoom: document.getElementById("quietRoom").checked,
      featherFreePillows: document.getElementById("featherFreePillows").checked,
      accessibleRoom: document.getElementById("accessibleRoom").checked,
      kingBed: document.getElementById("kingBed").checked,
      preferenceNotes: document.getElementById("preferenceNotes").value.trim()
    };

    if (!payload.firstName || !payload.lastName || !payload.email) {
      alert("Please fill in First Name, Last Name, and Email Address.");
      return;
    }

    fetch("/api/guest-registration/create", {
      method: "POST",
      headers: { "Content-Type": "application/json" },
      body: JSON.stringify(payload)
    })
      .then(response => response.json())
      .then(data => {
        if (data.success) {
          document.getElementById("guestId").value = data.guestId;
          document.getElementById("memberSince").value = data.memberSince;
          showGuestMessage(`Guest account created successfully. Guest ID: ${data.guestId}`);
        } else {
          alert("Unable to create guest account.");
        }
      });
  });

  clearBtn.addEventListener("click", function () {
    form.reset();
    loadGuestRegistrationMeta();
    removeGuestMessage();
  });
});

function loadGuestRegistrationMeta() {
  fetch("/api/guest-registration/meta")
    .then(response => response.json())
    .then(data => {
      document.getElementById("guestId").value = data.guestId;
      document.getElementById("memberSince").value = data.memberSince;
    });
}

function showGuestMessage(message) {
  removeGuestMessage();
  const msg = document.createElement("div");
  msg.className = "guest-message";
  msg.id = "guestMessage";
  msg.textContent = message;
  document.getElementById("guestRegistrationForm").appendChild(msg);
}

function removeGuestMessage() {
  const existing = document.getElementById("guestMessage");
  if (existing) existing.remove();
}

File: guest_registration.py
from database import get_connection, get_today_member_since, generate_guest_id


def get_guest_meta():
    return {
        "guestId": generate_guest_id(),
        "memberSince": get_today_member_since()
    }


def create_guest(data):
    guest_id = generate_guest_id()
    member_since = get_today_member_since()

    conn = get_connection()
    conn.execute("""
        INSERT INTO guests (
            guest_id, member_since, first_name, last_name, email, phone_number,
            date_of_birth, passport_id, high_floor, quiet_room, feather_free_pillows,
            accessible_room, king_bed, preference_notes
        )
        VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)
    """, (
        guest_id,
        member_since,
        data.get("firstName", ""),
        data.get("lastName", ""),
        data.get("email", ""),
        data.get("phoneNumber", ""),
        data.get("dateOfBirth", ""),
        data.get("passportId", ""),
        1 if data.get("highFloor") else 0,
        1 if data.get("quietRoom") else 0,
        1 if data.get("featherFreePillows") else 0,
        1 if data.get("accessibleRoom") else 0,
        1 if data.get("kingBed") else 0,
        data.get("preferenceNotes", "")
    ))
    conn.commit()
    conn.close()

    return {
        "success": True,
        "guestId": guest_id,
        "memberSince": member_since
    }